src/dial.js

```diff
@@ -87,8 +87,7 @@ module.exports = function dial (swarm) {
       nextTransport(tKeys.shift())
 
       function nextTransport (key) {
-        const multiaddrs = pi.multiaddrs.slice()
-        swarm.transport.dial(key, multiaddrs, (err, conn) => {
+        swarm.transport.dial(key, pi, (err, conn) => {
```

Comment:
Hm, I intentionally had made `swarm.transport.dial` be just multiaddr aware for simplicity, and so, it would only receive a multiaddr and not a peerInfo

swarm.transport.dial -> deals with dialing on a transport
swarm.dial -> does all the magic of upgrading the connection